src/Illuminate/Http/Request.php

```diff
@@ -255,6 +255,24 @@ public function has($key)
 	}
 
 	/**
+	 * Determine if the request contains a non-empty value for an input item matching a given pattern.
+	 *
+	 * @param  string  $pattern
+	 * @return bool
+	 */
+	public function hasRegex($pattern)
+	{
+		$input = $this->all();
+
+		foreach ($input as $key => $value)
```

Comment:
@phpeek is right.

[This](https://github.com/KennedyTedesco/framework/blob/input-l5/src/Illuminate/Http/Request.php#L229) has to be cached, so that it's not called again on every iteration in the loop. In our case it'll only pull it once, so there's no need for the `$input` variable.